python/dgl/distributed/graph_partition_book.py

```diff
@@ -830,6 +850,43 @@ def num_partitions(self):
         """
         return self._num_partitions
 
+    def init_canonical_etypes(self, etypes):
+        ''' initialize canonical etypes and related mapping.
+        '''
+        for etype, etype_id in etypes.items():
+            assert len(etype) == 3, \
+                'Canonical etype should be in format of (str, str, str).'
+            c_etype = etype
+            etype = etype[1]
+            self._etypes[etype_id] = etype
+            self._canonical_etypes[etype_id] = c_etype
+            if etype in self._etype2canonical:
+                # If one etype maps to multiple canonical etypes, empty
+                # tuple is used to indicate such ambiguity casued by etype.
+                # See more details in self.to_canonical_etype().
+                self._etype2canonical[etype] = tuple()
+            else:
+                self._etype2canonical[etype] = c_etype
+
+    @property
+    def init_by_canonical_etypes(self):
+        """ indicates whether partition book is initialized by canonical etypes.
+
+        .. deprecated:: 0.9.1
+            This function is deprecated as canonical etypes are always availale
+            for partitions generated since ``0.9.1``.
+        """
+        return self._init_by_canonical_etypes
+
+    def _etype_to_key(self, etype):
+        ''' obtain the key used in internal dict such as ``_typed_eid_range``.
```

Comment:
Don't understand why this is needed. The name `key` is ambiguous and `_typed_eid_range` has no explanation.

But the more fundamental issue is the overcomplication of the design. IMO, `GraphPartitionBook` is more an internal class that stores the meta information of partitions. In most cases, users should not see it. As such, we shouldn't be bothered with the burden to make it backward compatible. Please contact @zheng-da to see if their codebase is directly using `GraphPartitionBook` for some purposes.

Reply:
this is needed because we need to support parse of to-be-deprecated `meta.json`, be backward compatible. And such compatibility is not related to the exposure of `GPB` to user. Burden mainly comes from the support for both etype and canonical etypes.